reporter/src/main/scala/scoverage/reporter/CoberturaXmlWriter.scala

```diff
@@ -15,11 +15,22 @@ import scoverage.domain.MeasuredPackage
 class CoberturaXmlWriter(
     sourceDirectories: Seq[File],
     outputDir: File,
-    sourceEncoding: Option[String]
-) extends BaseReportWriter(sourceDirectories, outputDir, sourceEncoding) {
+    sourceEncoding: Option[String],
+    recoverNoSourceRoot: BaseReportWriter.PathRecoverer
```

Comment:
So I think we'll need to handle this a bit differently. These would be breaking changes for all the reporters, meaning another major bump, and I don't think we want to do another major atm. Can we potentially switch these up to either have a default implementation for this or another way that won't break compat for existing tools that are using this?

Reply:
I understand.
I can make the default throw an exception (i.e. keep the old behavior) and allow the "recoverer" to be added up.